Runtime/SpriteSwitchMixerBehaviour.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Playables;
+
+namespace Unity.MaterialSwitch
+{
+    internal class SpriteSwitchMixerPlayableBehaviour : PlayableBehaviour
+    {
+        private Dictionary<SpriteRenderer, MaterialPropertyBlock> _propertyBlocks = new Dictionary<SpriteRenderer, MaterialPropertyBlock>();
+        private static readonly int MainTex = Shader.PropertyToID("_MainTex");
+
+        public override void ProcessFrame(Playable playable, FrameData info, object playerData)
+        {
+            var group = playerData as SelectionGroups.SelectionGroup;
+            if (group == null) return;
+            if (!group.TryGetComponent(out SpriteGroup spriteGroup))
+            {
+                spriteGroup = group.gameObject.AddComponent<SpriteGroup>();
+            }
+            
+            // This code is disabled until we are sure it is not required.
+            // if (Application.isEditor && !Application.isPlaying)
+            // {
+            //     spriteGroup.CollectSpriteRenderers();
+            // }
+            
+            var inputCount = playable.GetInputCount();
+
+            var maxWeight = float.MinValue;
+            var maxIndex = -1;
+            //get total weight of all playables that are currently being mixed.
+            var totalWeight = 0f;
+            for (var i = 0; i < inputCount; i++)
```

Comment:
We are not blending sprites, so we can remove the entire weight calculation here.
There is a `ClipCaps` enum that you can use to prevent blending.